File: python/字符串/1496. 判断路径是否相交.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        s =set()
        s.add((0,0))
        x,y=0,0
        for i in path:
            if i == "N":
                y+=1
            elif i == "E":
                x+=1
            elif i == "S":
                y-=1
            elif i == "W":
                x-=1
            if (x,y) in s:
                return True
            s.add((x,y))
        return False

File: go/string/1496. 判断路径是否相交.go
package string

type direction struct {
	x int
	y int
}

func isPathCrossing(path string) bool {
	s := make(map[direction]struct{})
	d := direction{}
	s[d] = struct{}{}
	for _, v := range path {
		switch v {
		case 'N':
			d = direction{d.x, d.y + 1}
		case 'S':
			d = direction{d.x, d.y - 1}
		case 'E':
			d = direction{d.x + 1, d.y}
		case 'W':
			d = direction{d.x - 1, d.y}
		}
		if _, ok := s[d]; ok {
			return true
		}
		s[d] = struct{}{}

	}
	return false
}
